.github/workflows/main.yml

```diff
@@ -25,3 +25,18 @@ jobs:
       run: make benchmark
     - name: Run benchmark - WAMR
       run: make benchmark.wamr
+  check-macos:
```

Comment:
you could probably avoid duplicating the code here and use a [matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) instead:
```yml
 strategy:
      matrix:
        platform: [linux, macos]
```
and then use it in the job itself:
```
    - name: Build components
      run: make PLATFORM=${{ matrix.platform }}
```